File: src/test/com/intellij/lang/jsgraphql/validation/GraphQLSchemaValidationTest.java
package com.intellij.lang.jsgraphql.validation;

import com.intellij.lang.jsgraphql.GraphQLTestCaseBase;
import org.jetbrains.annotations.NotNull;

public class GraphQLSchemaValidationTest extends GraphQLTestCaseBase {
    @Override
    protected @NotNull String getBasePath() {
        return "/validation/schema";
    }

    @Override
    protected void setUp() throws Exception {
        super.setUp();
        enableAllInspections();
    }

    public void testTypeRegistryRedefinitionErrors() {
        doHighlightingTest();
    }

    public void testNotInputOutputTypesErrors() {
        doHighlightingTest();
    }

    public void testSchemaValidatorErrors() {
        doHighlightingTest();
    }

    public void testRedefinitionErrors() {
        doHighlightingTest();
    }

    public void testImplementingErrors() {
        doHighlightingTest();
    }

    public void testImplementingArgumentsErrors() {
        doHighlightingTest();
    }

    public void testEnumErrors() {
        doHighlightingTest();
    }

    public void testDirectiveErrors() {
        doHighlightingTest();
    }

    public void testSchemaDirectiveErrors() {
        doHighlightingTest();
    }

    public void testObjectSchemaTypeExtensionsErrors() {
        doHighlightingTest();
    }

    public void testInterfaceSchemaTypeExtensionsErrors() {
        doHighlightingTest();
    }

    public void testUnionSchemaTypeExtensionsErrors() {
        doHighlightingTest();
    }

    public void testEnumSchemaTypeExtensionsErrors() {
        doHighlightingTest();
    }

    public void testScalarSchemaTypeExtensionsErrors() {
        doHighlightingTest();
    }

    public void testInputSchemaTypeExtensionsErrors() {
        doHighlightingTest();
    }

    public void testRepeatableDirectivesErrors() {
        doHighlightingTest();
    }

    public void testExtendSchemaErrors() {
        doHighlightingTest();
    }

    public void testFieldDefinitionDirectiveErrors() {
        doHighlightingTest();
    }

    public void testSchemaOperationTypeErrors() {
        doHighlightingTest();
    }

    public void testDeprecatedMembers() {
        doHighlightingTest();
    }

}

File: tests/testSrc/com/intellij/lang/jsgraphql/validation/GraphQLSchemaValidationTest.kt
package com.intellij.lang.jsgraphql.validation

import com.intellij.lang.jsgraphql.GraphQLTestCaseBase
import com.intellij.openapi.progress.runBlockingCancellable

class GraphQLSchemaValidationTest : GraphQLTestCaseBase() {
  override fun getBasePath(): String {
    return "/validation/schema"
  }

  override fun setUp() {
    super.setUp()
    enableAllInspections()
  }

  fun testTypeRegistryRedefinitionErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testNotInputOutputTypesErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testSchemaValidatorErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testRedefinitionErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testBuiltInRedefinitionErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testImplementingErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testImplementingArgumentsErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testEnumErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testDirectiveErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testSchemaDirectiveErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testObjectSchemaTypeExtensionsErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testInterfaceSchemaTypeExtensionsErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testUnionSchemaTypeExtensionsErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testEnumSchemaTypeExtensionsErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testScalarSchemaTypeExtensionsErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testInputSchemaTypeExtensionsErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testRepeatableDirectivesErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testExtendSchemaErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testFieldDefinitionDirectiveErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testSchemaOperationTypeErrors() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testDeprecatedMembers() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testInterfaceImplementationWithFieldSubType() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testDuplicatedTypeWithDescription() = runBlockingCancellable {
    doHighlightingTest()
  }
}
